vaadin-spring/src/main/java/com/vaadin/flow/spring/SpringHelper.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2000-2017 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.spring;
+
+import java.util.Iterator;
+import java.util.Locale;
+
+import org.springframework.core.env.AbstractEnvironment;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.core.env.PropertySource;
+
+/**
+ * Utility methods for spring
+ */
+public class SpringHelper {
+
+    private SpringHelper() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    /**
+     * Gets the property name used after being parsed by Spring Boot, accepting
+     * relaxed parsing
+     *
+     * @param env the environment
+     * @param propertyName the relaxed property name
+     * @return the actual property name
+     */
+    public static String getActualPropertyName(Environment env, String propertyName) {
+        String storedPropertyName = null;
+        String normalizedPropertyName = normalizePropertyName(propertyName);
+        Iterator it = ((AbstractEnvironment) env).getPropertySources().iterator();
+        while(it.hasNext() && storedPropertyName == null) {
+            PropertySource propertySource = (PropertySource) it.next();
+            if (propertySource instanceof MapPropertySource) {
+                storedPropertyName = extractFromPropertySource(
+                        propertySource, normalizedPropertyName);
+            }
+        }
+        return storedPropertyName;
+    }
+
+    private static String extractFromPropertySource(
+            PropertySource propertySource, String normalizedPropertyName) {
+        String storedPropertyName = null;
+        for (String key : ((MapPropertySource) propertySource)
+                .getSource().keySet()) {
+            if (normalizePropertyName(key).equals(normalizedPropertyName)) {
```

Comment:
Normalization process may results in unexpected matches.

E.g. `foo__bar` will match `foo_bar`.
Is it OK ?

Reply:
for me it is a valid result. I just want to check that they go to the same place. Also, the method is only checking vaadin properties, so I think it is acceptable to think that no conflicts will happen

Reply:
This becomes kind of internal API.
So anyone who updates Spring code may use this method .
The contract doesn't limit anyone from using the method only for Vaadin properties which are hardcoded.

The result is not valid for me generally.
It has a lot of issues :
- `foo.bar` property is normalized as `foobar` which means the property `bar` in the `foo` namespace is the same as `foobar` which is incorrect
- property `foo__bar` is the same as `foobar`.
- property `foo--bar` is the same as `foobar`.

Neither _existing constant_ Vaadin property has `_` /`-` inside the name.
So it's not a Vaadin property but rather something which _may_ become a Vaadin property. And the examples  shown above may convert a property which should not be a Vaadin property into Vaadin property.

The last concern: I cannot find the usages of `SpringHelper` class except the only one inside `RootMappedCondition`.
Am I missing anything ?
I don't see any point to have such method for only one usecase which can be handled properly for only one property.